apis/python/src/tiledbsoma/_read_iters.py

```diff
@@ -5,59 +5,424 @@
+        """
+        Unimplemented as there is little utility beyond that offered by a ragged
+        read iterator concat, other than reindexing.
+        """
+        raise NotImplementedError("Blockwise iterators do not support concat operation")
+
+    _EagerRT = TypeVar("_EagerRT")
+
+    def _maybe_eager_iterator(
+        self, x: Iterator[_EagerRT], _pool: Optional[ThreadPoolExecutor] = None
+    ) -> Iterator[_EagerRT]:
+        """Private"""
+        return EagerIterator(x, pool=_pool) if self.eager else x
+
+    def _table_reader(self) -> BlockwiseSingleAxisTableIter:
+        """Private. Blockwise table reader. Helper function for sub-class use"""
+        kwargs: Dict[str, object] = {"result_order": self.sr.result_order}
+        for coord_chunk in _coords_strider(
+            self.coords[self.major_axis],
+            self.sr.shape[self.major_axis],
+            self.size[0],
+        ):
+            self.sr.reset(**kwargs)
+            step_coords = list(self.coords)
+            step_coords[self.major_axis] = coord_chunk
+            self.array._set_reader_coords(self.sr, step_coords)
 
+            joinids = list(self.joinids)
+            joinids[self.major_axis] = pa.array(coord_chunk)
+            yield pa.concat_tables(_arrow_table_reader(self.sr)), tuple(joinids)
 
-class SparseTensorReadIterBase(somacore.ReadIter[RT], metaclass=abc.ABCMeta):
+    def _reindexed_table_reader(
+        self, _pool: Optional[ThreadPoolExecutor] = None
+    ) -> BlockwiseSingleAxisTableIter:
+        """Private. Blockwise table reader w/ reindexing. Helper function for sub-class use"""
+        for tbl, coords in self._maybe_eager_iterator(self._table_reader(), _pool):
+            pytbl = {}
+            for d in range(self.ndim):
+                col = tbl.column(f"soma_dim_{d}")
+                if d in self.axes_to_reindex:
+                    if d == self.major_axis:
+                        col = pd.Index(coords[self.major_axis]).get_indexer(col.to_numpy())  # type: ignore[no-untyped-call]
+                    else:
+                        col = self.minor_axes_indexer[d].get_indexer(col.to_numpy())  # type: ignore[no-untyped-call]
+                pytbl[f"soma_dim_{d}"] = col
+            pytbl["soma_data"] = tbl.column("soma_data")
+            yield pa.Table.from_pydict(pytbl), coords
+
+
+class BlockwiseTableReadIter(BlockwiseReadIterBase[BlockwiseTableReadIterResult]):
+    """Blockwise iterator over `Arrow Table <https://arrow.apache.org/docs/python/generated/pyarrow.Table.html>`_ elements"""
+
+    def _create_reader(self) -> Iterator[BlockwiseTableReadIterResult]:
+        """Private. Blockwise Arrow Table iterator, restricted to a single axis"""
+        if self.eager and not self.eager_iterator_pool:
+            with ThreadPoolExecutor() as _pool:
+                yield from self._reindexed_table_reader(
+                    _pool
+                ) if self.axes_to_reindex else self._table_reader()
```

Comment:
These indentations are extremely weird. Maybe

```python
yield from (
    self._reindexed_whatever(...)
    if ... else ...
)
```
if it will let you?

Reply:
it was whatever ruff/black put in place.  I will try your proposal - it should work fine.